x/gov/types/msgs.go

```diff
@@ -290,3 +265,21 @@ func (msg MsgVoteWeighted) GetSigners() []sdk.AccAddress {
 	voter, _ := sdk.AccAddressFromBech32(msg.Voter)
 	return []sdk.AccAddress{voter}
 }
+
+func NewCoins(coins []*sdk.Coin) sdk.Coins {
+	var c []sdk.Coin
+	for _, coin := range coins {
+		if coin != nil {
+			c = append(c, *coin)
+		}
+	}
+	return sdk.NewCoins(c...)
+}
+
+func ToCoinSlice(coins sdk.Coins) []*sdk.Coin {
+	slice := make([]*sdk.Coin, len(coins))
+	for idx, coin := range coins {
+		slice[idx] = &coin
+	}
+	return slice
+}
```

Comment:
I had to create these conversion functions as a result of dropping gogoproto so the proto structs are now an array of *sdk.Coin

Reply:
IMO it makes the code harder to understand.

In all other modules we generate into `[]types.Coin`, and never into `[]*types.Coin`. For consistency, I would still use gogoproto.nullable. We can then remove these 2 converters.

We should eventually remove gogoproto.nullable for coins for sure, but my preference is to do it in one go for all modules.

Reply:
Ok I will add gogoproto back